packages/liferay-npm-scripts/test/jsp/formatJSP.js

```diff
@@ -88,6 +88,115 @@ describe('formatJSP()', () => {
 		expect(formatJSP(source)).toBe(expected);
 	});
 
+	describe('fixing problems with indentation relative to script tag (#437)', () => {
+		// ie. each line is correct with respect to its neighbors, but overall,
+		// the code is wrong relative to the script tag.
+
+		it('corrects script content that is insufficiently indented', () => {
+			const source = `
+				<aui:script>
+				function <portlet:namespace />deleteOrganization(
```

Comment:
I haven't checked the code, I'm sure it's likely covered and that it's highly unlikely we have code like this anywhere... but have you checked with negatively indented code?

```jsp
        <aui:script>
function foo() {
}
        </aui:script>
```

Reply:
And by I haven't checked I mean I can barely understand it

Reply:
I haven't checked it but I'm sure it works. I'll add a test.

Reply:
Negative indent test: https://github.com/liferay/liferay-npm-tools/pull/468/commits/0ed44061c85fb697878f4282c00802dc2bc5c878

Reply:
> I can barely understand it just by looking over it briefly

Added some explanation in https://github.com/liferay/liferay-npm-tools/pull/468/commits/77351cebc8a099b400fb1704ba602e18978c06cc which might help?

I think the `baseIndent` calculation is the important bit.

The tricky dealing with `prefix` and `suffix` is another story that I can only explain via Jedi mind trick. Nothing to see there... move along...

![not-the-droids](https://user-images.githubusercontent.com/7074/85050353-40e23a00-b196-11ea-8574-67868e985f54.gif)

Reply:
> [...] plus 1 because columns are
1-indexed [...]

This is pure evil 😂

Reply:
Are you saying there's a Sith connection here? It's pretty standard for editors/IDE and tools that operate on source (eg. ESLint, Babel etc) to be 1-based.

![evil-sith](https://user-images.githubusercontent.com/7074/85121168-98c78200-b224-11ea-9ff2-fdd38f8dd6d8.gif)

Reply:
> It's pretty standard for editors/IDE [...]

You would know, you VIM lover!

Reply:
I once had to open VSCode to help a Flash developer, so I have some experience with that too... 😂